src/main/java/io/airlift/slice/SizeOf.java

```diff
@@ -95,6 +101,21 @@ public static long estimatedSizeOf(String string)
         return (string == null) ? 0 : (STRING_INSTANCE_SIZE + string.length() * Character.BYTES);
     }
 
+    public static <K, V> long estimatedSizeOf(Map<K, V> map, ToLongFunction<K> keySize, ToLongFunction<V> valueSize)
+    {
+        if (map == null) {
+            return 0;
+        }
+
+        long result = sizeOfObjectArray(map.size());
+        for (Entry<K, V> entry : map.entrySet()) {
+            result += SIMPLE_ENTRY_INSTANCE_SIZE +
```

Comment:
this is where this differs from at least some places where we estimate map size in Presto

Reply:
Maybe we should have a function for this, so we can compute entry size based on the implementation. Would Guava immutable maps often be used with this method? What implementations are common?

Reply:
unclear to me, what would be your suggestion?